tools/cli.js

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env node
+
+'use strict';
+
+// TODO: support jstp://server and jstps://server
+
+const jstp = require('.');
+const readline = require('readline');
+
+const commandProcessor = {};
+const lineProcessor = {};
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+  completer
+});
+
+const log = msg => {
+  const userInput = rl.line;
+  if (userInput) rl.clearLine();
+  console.log(msg);
+  rl.write('\n');
+  if (userInput) rl.write(userInput);
+};
+
+rl.on('line', (line) => {
+  const [type, leftover] = _split(line.trim(), ' ', 1);
+  if (!type) {
+    return rl.prompt(true);
+  }
+  const processor = lineProcessor[type];
+  if (!processor) {
+    log(`Unknown command ${type}`);
+  } else {
+    processor(leftover, (err, result) => {
+      if (err) return log(`${err.name} occurred: ${err.message}`);
+      log(result);
+    });
+    rl.prompt(true);
+  }
+});
+
+rl.on('SIGINT', () => {
+  rl.close();
+  process.exit();
+});
+
+rl.on('close', () => {
+  process.exit();
+});
+
+function completer(line) {
+  return [[], line];
+}
+
+const state = {
+  client: null,
+  connection: null
+};
+
+commandProcessor.call = (interfaceName, methodName, args, callback) => {
+  if (state.client === null) return callback(new Error('Not connected'));
+  state.connection.callMethod(interfaceName, methodName, args, callback);
+};
+
+commandProcessor.event = (interfaceName, eventName, args, callback) => {
+  if (state.client === null) return callback(new Error('Not connected'));
+  state.connection.emitRemoteEvent(interfaceName, eventName, args);
+  callback();
+};
+
+commandProcessor.connect = (host, port, appName, callback) => {
+  state.client = jstp.tcp.createClient({ host, port, secure: true });
+  state.client.connectAndHandshake(appName, null, null,
+      (err, connection) => {
+        if (err) return callback(err);
+        state.connection = connection;
+        // todo make event registering generic
```

Comment:
The proper spelling of `todo` is `TODO:` :) That'll make it highlighted in code editors and picked up to be shown on the bitHound dashboard.